Assets/__Scripts/MapEditor/UI/CurrentSectionDisplay.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using TMPro;
+using UnityEngine;
+
+[RequireComponent(typeof(TextMeshProUGUI))]
+public class CurrentSectionDisplay : MonoBehaviour
+{
+    [SerializeField] private AudioTimeSyncController atsc;
+    [SerializeField] private BookmarkManager bookmarkManger;
+    private TextMeshProUGUI textMesh;
+
+    private Stack<BookmarkContainer> upcomingBookmarks = new Stack<BookmarkContainer>();
+    private bool isPlaying;
+
+    void Awake()
+    {
+        textMesh = GetComponent<TextMeshProUGUI>();
+    }
+
+    void OnEnable()
+    {
+        atsc.OnTimeChanged += OnTimeChanged;
+        atsc.OnPlayToggle += OnPlayToggle;
+    }
+
+    void OnDisable()
+    {
+        atsc.OnTimeChanged -= OnTimeChanged;
+        atsc.OnPlayToggle -= OnPlayToggle;
+    }
+
+    private void OnTimeChanged()
+    {
+        if (isPlaying)
+        {
+            if (upcomingBookmarks.Count == 0)
+                return;
+            if (upcomingBookmarks.Peek().data._time <= atsc.CurrentBeat)
+                textMesh.text = upcomingBookmarks.Pop().data._name;
+        }
+        else
+        {
+            textMesh.text = bookmarkManger.bookmarkContainers.LastOrDefault(x => x.data._time <= atsc.CurrentBeat)?.data._name ?? "";
```

Comment:
Very small performance nitpick: `.Where(predicate).LastOrDefault()` has been proven to be faster than `.LastOrDefault(predicate)`, but since this only happens when the user is not playing the song, I guess it's not *too* bad.